Tests/UnitTests/Purchasing/Purchases/PurchasesLogInTests.swift

```diff
@@ -113,6 +113,30 @@ class PurchasesLogInTests: BasePurchasesTests {
         expect(self.identityManager.invokedSwitchUserParametersList) == ["test-user-id"]
     }
 
+    func testRefreshesOfferingsCache() {
+        expect(self.mockOfferingsManager.invokedUpdateOfferingsCacheCount) == 0
+
+        self.systemInfo = MockSystemInfo(finishTransactions: true, customEntitlementsComputation: true)
+        Purchases.clearSingleton()
+        self.initializePurchasesInstance(appUserId: "old-test-user-id")
+
+        self.purchases.switchUser(to: "test-user-id")
+
+        expect(self.mockOfferingsManager.invokedUpdateOfferingsCacheCount).toEventually(equal(1))
+    }
+
+    func testSwitchUserNoOpIfAppUserIDIsSameAsCurrent() {
+        self.systemInfo = MockSystemInfo(finishTransactions: true, customEntitlementsComputation: true)
+        Purchases.clearSingleton()
+        let appUserId = "test-user-id"
+        self.initializePurchasesInstance(appUserId: appUserId)
+
+        self.purchases.switchUser(to: appUserId)
+
+        expect(self.identityManager.invokedSwitchUser) == false
+        expect(self.mockOfferingsManager.invokedUpdateOfferingsCacheCount).toEventually(equal(0))
```

Comment:
This eventually isn't needed, since it'll be 0 to begin with, right?

Reply:
exactly the point, though: it'll be zero at first, but if there was a bug, it'd happen asynchronously. So it'd always be zero if you check right away, you need to check a little bit later to ensure that it wasn't triggered

Reply:
But this `toEventually` will only check once synchronously, pass, and then not check again :)

Reply:
oh lol right

Reply:
updated